pkg/pillar/containerd/containerd.go

```diff
@@ -403,7 +403,11 @@ func CtrContainerInfo(name string) (int, string, error) {
 		return 0, "", fmt.Errorf("CtrContainerInfo: couldn't determine task status for container %s: %v", name, err)
 	}
 
-	return int(t.Pid()), string(stat.Status), nil
+	if stat.Status == "stopped" {
+		return int(stat.ExitStatus), string(stat.Status), nil
+	} else {
```

Comment:
The overloading of the int return value with the existstatus might warrant a comment here and in the calle

Reply:
Yup. That's what's happening ;-)

Now, on one hand -- I agree -- it is not clean. On the other hand -- when task is HALTED the PID is no longer meaningful so we may as well re-use that return value. So in short -- what we *really* need here is a union type to be returned (but I don't think there's a clean way to do that in Go).

I'm open to other suggestions, of course -- would you prefer this to be an extra return argument? Would you prefer we simply return stat and let the caller parse it out and see what's inside of it?

Reply:
An extra return arg would be a lot more obvious.

Reply:
Fixed